graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/modules/BuiltinFunctions.java

```diff
@@ -286,6 +294,209 @@ public Object absObject(VirtualFrame frame, Object object,
+                    return true;
+                }
+            }
+
+            return nodeType == NodeType.ALL;
+        }
+    }
+
+    @Builtin(name = ALL, minNumOfPositionalArgs = 1)
+    @GenerateNodeFactory
+    public abstract static class AllNode extends PythonUnaryBuiltinNode {
+
+        @Specialization(guards = "cannotBeOverridden(object, getClassNode)", limit = "1")
+        boolean doList(VirtualFrame frame,
+                        PList object,
+                        @SuppressWarnings("unused") @Shared("getClassNode") @Cached GetClassNode getClassNode,
+                        @Shared("allOrAnyNode") @Cached AllOrAnyNode allOrAnyNode) {
+            return allOrAnyNode.execute(frame, object.getSequenceStorage(), AllOrAnyNode.NodeType.ALL);
+        }
+
+        @Specialization(guards = "cannotBeOverridden(object, getClassNode)", limit = "1")
+        boolean doTuple(VirtualFrame frame,
+                        PTuple object,
+                        @SuppressWarnings("unused") @Shared("getClassNode") @Cached GetClassNode getClassNode,
+                        @Shared("allOrAnyNode") @Cached AllOrAnyNode allOrAnyNode) {
+            return allOrAnyNode.execute(frame, object.getSequenceStorage(), AllOrAnyNode.NodeType.ALL);
+        }
+
+        @Specialization(guards = "cannotBeOverridden(object, getClassNode)", limit = "1")
+        boolean doDict(VirtualFrame frame,
+                        PDict object,
+                        @SuppressWarnings("unused") @Shared("getClassNode") @Cached GetClassNode getClassNode,
+                        @Shared("allOrAnyNode") @Cached AllOrAnyNode allOrAnyNode) {
+            return allOrAnyNode.execute(frame, object.getDictStorage(), AllOrAnyNode.NodeType.ALL);
+        }
+
+        @Specialization(guards = "cannotBeOverridden(object, getClassNode)", limit = "1")
+        boolean doSet(VirtualFrame frame,
+                        PBaseSet object,
+                        @SuppressWarnings("unused") @Shared("getClassNode") @Cached GetClassNode getClassNode,
+                        @Shared("allOrAnyNode") @Cached AllOrAnyNode allOrAnyNode) {
+            return allOrAnyNode.execute(frame, object.getDictStorage(), AllOrAnyNode.NodeType.ALL);
+        }
+
+        @Specialization
+        boolean doObject(VirtualFrame frame,
+                        Object object,
+                        @Cached PyObjectGetIter getIter,
+                        @Cached GetNextNode nextNode,
+                        @Cached IsBuiltinClassProfile errorProfile,
+                        @Cached PyObjectIsTrueNode isTrueNode) {
+            Object iterator = getIter.execute(frame, object);
+            while (true) {
+                try {
+                    Object next = nextNode.execute(frame, iterator);
+                    if (!isTrueNode.execute(frame, next)) {
+                        return false;
+                    }
+                } catch (PException e) {
```

Comment:
Optional: You could report the execution count of such `while(true)` loops using `LoopNode.reportLoopCount`.

Reply:
Would that not require making the `doObject` specialization methods not static, to be able to pass a `this` pointer to the method?

Reply:
Yes, it would. Then don't make this one `static`. We just try to make methods `static` when they don't use `this` in any way, but it's not a problem at all to have them as instance methods when they do.

Reply:
Implemented in de15e38, with a call to it in `AllOrAnyNode.doHashStorage` as well (it used a `for` loop over keys of a `HashingStorage` object) ; it isn't a `while(true)` loop, and therefore I'm not sure if I should revert that specific change.